Modules/_collectionsmodule.c

```diff
@@ -2572,14 +2572,7 @@ collections_exec(PyObject *module) {
     defdict_type.tp_base = &PyDict_Type;
 
     for (int i = 0; typelist[i] != NULL; i++) {
```

Comment:
What do you think of using `Py_ARRAY_LENGTH()` instead of NULL terminator?

```suggestion
    for (int i = 0; i < Py_ARRAY_LENGTH(typelist); i++) {
```

If you like it, please also modify other files.

Reply:
Got it!